pysph/sph/scheme.py

```diff
@@ -120,7 +123,11 @@ def _ensure_properties(self, pa, desired_props, clean=True):
 
         to_add = all_props - set(pa.properties.keys())
         for prop in to_add:
-            pa.add_property(prop)
+            if stride_info is not None and prop in stride_info:
```

Comment:
Maybe it would be better to pass in the desired_props as a sequence or strings or dicts.  If its a dict, you use it as `add_property(**item)` else as just `add_property(item)`.

Reply:
changed it to dict, also added few lines which ensure that the desired_props have no duplicates even when the user does the wrong thing.